### PR TITLE
[Fix] sampling_params and Runtime bugs for some cases.

### PR DESCRIPTION
**Pull request by gryffindor-rr**
## Modification
bug 1:
data = json.dumps(sampling_params) and then SamplingPramas(json.loads(data)), will report no attribute 'stop_strs', modify the position argument name consistent to self variable.

bug 2:
sampling_params { "n": 3,...} will report error, 'Do not support stream for batch mode'
Fix runtime to set stream when n>1.

Tested by examples/usage/async_io.py
## Checklist

- [x] Before submitting a PR for review, make sure it has passed verification in your local development environment **at least**.
- [x] Ensure pre-commit `pre-commit run --all-files` or other linting tools are used to fix potential lint issues.
- [x] Confirm that modifications are covered by complete unit tests. If not, please add more unit tests for correctness.
- [x] Modify documentation as needed, such as docstrings or example tutorials.

### CODE REVIEW

**Comment by vhain** [comment]
`stream: true` with `n > 1` could be supported with https://github.com/sgl-project/sglang/pull/1134

**Comment by gryffindor-rr** [comment]
or maybe another esier way to support json.dumps for sampling_params: https://github.com/gryffindor-rr/sglang/commit/de4be50de34d741dea550d6e6f10c8bcd966bbd4

File: python/sglang/srt/server.py [hunk 1]
```diff
@@ -521,17 +521,22 @@ async def async_generate(
         prompt: str,
         sampling_params: Optional[Dict] = None,
     ):
+        if isinstance(sampling_params, dict) and sampling_params.get("n", 1) > 1:
+            stream = False
+        else:
+            stream = True
```
Comment (merrymercy) [inline python/sglang/srt/server.py:527]: Revert this as it is supported in https://github.com/sgl-project/sglang/commit/d8476818efc88188d0aa0a8a176024a0b82e7a1d

**Review by merrymercy — CHANGES_REQUESTED** [review]
Given that `stop` is in the public API, we do not want to change it to break existing apps.

Your new commit is better. Can you submit a PR for that?

**Comment by merrymercy** [comment]
Close this in favor of https://github.com/gryffindor-rr/sglang/commit/de4be50de34d741dea550d6e6f10c8bcd966bbd4

**Pull request closed without merging**